TOOLS/umpv

```diff
@@ -26,40 +26,34 @@ Note: you can supply custom mpv path and options with the MPV environment
       is already running), this will be ignored.
 """
 
-import sys
+import errno
 import os
+import shlex
 import socket
-import errno
-import subprocess
 import string
-import shlex
-
-files = sys.argv[1:]
+import subprocess
+import sys
 
 # this is the same method mpv uses to decide this
 def is_url(filename):
     parts = filename.split("://", 1)
     if len(parts) < 2:
         return False
     # protocol prefix has no special characters => it's an URL
-    allowed_symbols = string.ascii_letters + string.digits + '_'
+    allowed_symbols = string.ascii_letters + string.digits + "_"
     prefix = parts[0]
     return all(map(lambda c: c in allowed_symbols, prefix))
 
 # make them absolute; also makes them safe against interpretation as options
-def make_abs(filename):
-    if not is_url(filename):
-        return os.path.abspath(filename)
-    return filename
-files = (make_abs(f) for f in files)
+files = (os.path.abspath(f) if not is_url(f) else f for f in sys.argv[1:])
 
-SOCK = os.path.join(os.getenv("XDG_RUNTIME_DIR", os.getenv("HOME")), ".umpv_socket")
+SOCK = os.path.join(os.getenv("XDG_RUNTIME_DIR", os.getenv("HOME", "")), ".umpv_socket")
 
 sock = None
 try:
     sock = socket.socket(socket.AF_UNIX)
     sock.connect(SOCK)
-except socket.error as e:
+except OSError as e:
     if e.errno == errno.ECONNREFUSED:
         sock = None
```

Comment:
this actually leaks `sock`

Reply:
I can fix that, but I will likely rewrite this whole thing, because it is kinda iffy.

Reply:
Fixed.